README.md

```diff
@@ -1,13 +1,102 @@
 # Epochtal Tournament System
-Framework for autonomous management of Portal 2 tournaments.
 
-Reaching for the ambitious goal of bringing Portal 2 as close to a competitive esport as possible, starting by building a solid technical foundation. Intended for use in [epochtal.p2r3.com](https://epochtal.p2r3.com/), designed for adaptability in any Portal 2 tournament context.
+Epochtal is a framework for autonomous management of Portal 2 tournaments.
+
+Reaching for the ambitious goal of bringing Portal 2 as close to a competitive e-sport as possible, starting by building
+a solid technical foundation. Originally developed for use in [epochtal.p2r3.com](https://epochtal.p2r3.com/), designed
+for adaptability in any Portal 2 tournament context.
 
 ## Setup
+
 This project is written for the [Bun runtime](https://bun.sh/), and will not work on other runtimes such as Node or Deno.
 
-To set it up, install the dependencies and run the entrypoint file:
-```bash
-bun install
+There are several ways you can set up epochtal for your own tournament:
+
+### Using Docker (recommended)
+
+Epochtal builds and publishes a Docker image every time a changes are pushed to the `main` branch.
+
+To run epochtal through Docker compose, use the provided [docker-compose.yml](docker-compose.yml) file.
+
+If you want to use 'Plain Docker', you should first create a volume to persistently store data between updates:
+
+```shell
+docker volume create epochtal-data
+```
+
+Then, you can start the container:
+
+```shell
+docker run -d \
+-v epochtal-data:/app/data \
+-p 8080:8080 \
+-e STEAM_API_KEY=changeme \
+-e DISCORD_API_KEY=changeme \
+-e JWT_SECRET=changeme \
+-e INTERNAL_SECRET=changeme \
+-e DISCORD_CHANNEL_ANNOUNCE=changeme \
+-e DISCORD_CHANNEL_REPORT=changeme \
+-e DISCORD_CHANNEL_UPDATE=changeme \
+ghcr.io/p2r3/epochtal
+```
+
+### Manual installation
+
+If you don't want to use Docker, ~~use docker~~ you can run epochtal locally on your machine:
+
+1. Install the required dependencies; epochtal needs `tar`, `xz` and `wget` to be available to the Bun process.
+2. Clone the repository
+   ```shell
+   git clone --depth 1 https://github.com/p2r3/epochtal
+   cd epochtal
+   ```
+3. Install the necessary binaries:
+   ```shell
+   mkdir bin 
+   cd bin
+   # Install bspsrc
+   wget https://github.com/ata4/bspsrc/releases/download/v1.4.5/bspsrc-linux.zip
+   unzip bspsrc-linux.zip -d bspsrc
+   # Install mdp-json
+   mkdir mdp-json
+   wget https://github.com/p2r3/mdp-json/releases/download/07.2024/mdp -O mdp-json/mdp
+   chmod +x mdp-json/mdp
+   # Install UntitledParser
+   wget https://github.com/UncraftedName/UntitledParser/releases/download/jul-9-2023/UntitledParser-linux -O UntitledParser
+   chmod +x UntitledParser
+   ```
+4. Copy `.env.example` to `.env` and populate the necessary fields
+5. Install bun packages and run epochtal!
+   ```shell
+   bun install --production
+   bun run main.js
+   ```
```

Comment:
Should mention somewhere that --production skips development dependencies. It's obvious if you think about it but might still cause confusion (with something like a "release build" with compiler optimizations)

Reply:
Well, in this case that's just a good thing? I could probably mention it but it shouldn't matter - these instructions are for what you'd do if you want to deploy it, in which case you dont need devDependencies. (or am i missing something and it could be good to have devDependencies?)

Reply:
Oh I was thinking that "using docker" is how to deploy Epochtal and "manual installation" would be how to run Epochtal locally for development purposes, in which case --production would obviously not be required

Reply:
Oh yeah now that I think about it that's kinda unclear yeah. I meant everything as different methods of deployment lmao, most people that wanna contribute probably know how to setup bun